browser/src/screenshot.ts

```diff
@@ -0,0 +1,20 @@
+import { type Page } from 'playwright'
+import * as gptscript from '@gptscript-ai/gptscript'
+
+const client = new gptscript.GPTScript();
+
+export async function screenshot (page: Page): Promise<string> {
+  let screenshotName = `screenshot-${Date.now()}_${page.url().replace(/[^a-zA-Z0-9]/g, '_')}.png`
+  // Detect if we are running in otto8
+  if (process.env.OTTO_THREAD_ID !== undefined) {
+    screenshotName = `files/${screenshotName}`
+  }
+  const screenshot = await page.screenshot()
+
+  try {
+    await client.writeFileInWorkspace(screenshotName, screenshot)
```

Comment:
this method will default to the workspace ID environment variable if it's not given as an argument, but since this runs as a daemon that won't change after startup. IIUC, this means that all users will be saving their screenshots to the same workspace.

Instead, I think the workspace ID should be extracted from the request headers, [like we do here](https://github.com/otto8-ai/tools/blob/f80a15a6bdf066994d20747261bd657866896381/browser/src/session.ts#L121).